### PR TITLE
Deploy notary on staging.

### PR DESCRIPTION
**Pull request by RichardScothern**
Create a stage-config files without SSL keys as SSL is terminated
at the ELB.

Add a service name for correct routing from https://notary-stage.docker.io

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,4 +12,5 @@ EXPOSE 4443
 
 WORKDIR /go/src/github.com/docker/notary
 
-CMD [ "notary-server", "-config", "cmd/notary-server/dev-config.json" ]
+
+CMD [ "notary-server", "-config", "cmd/notary-server/stage-config.json" ]
```
Comment (NathanMcCauley) [inline Dockerfile]: Does this imply dockerfile per environment? Is that how the rest of the apps in our infra do it?

File: docker-compose.yml [hunk 2]
```diff
@@ -14,3 +16,4 @@ notarymysql:
   build: ./notarymysql/
   ports:
     - "3306:3306"
+
```
Comment (NathanMcCauley) [inline docker-compose.yml]: remove

File: Dockerfile (hunk 1 above)
Comment (thaJeztah) [inline Dockerfile]: Perhaps

```Dockerfile
CMD [ "--config", "cmd/notary-server/stage-config.json" ]
ENTRYPOINT [ "notary-server" ]
```

That way the config can be overridden from the command-line, e.g.

    docker run notary-server -config foobar.json

Comment (thaJeztah) [inline Dockerfile]: (unrelated) Just noticed that notary uses a single dash for non-shorthand options, i.e. `-options`, not `--options`? This is different from Docker itself, which uses `--` for non-shorthand options.

Comment (RichardScothern) [inline Dockerfile]: There currently isn't a way to pass runtime environment variables through docker-compose to the Dockerfile. (https://github.com/docker/compose/issues/1377).  I'll talk to infra.

@thaJeztah  '-config' is a byproduct of the command line library used by notary.

File: cmd/notary-server/stage-config.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+	"server": {
+		"addr": ":4443"
+	},
+	"trust_service": {
+		"type": "local",
```
Comment (diogomonica) [inline cmd/notary-server/stage-config.json]: This should probably not be local and should be pointed to notary-signer.

File: docker-compose.yml [hunk 4]
```diff
@@ -3,8 +3,10 @@ notary:
   links:
    - notarymysql
   ports:
-   - "8080:8080"
-   - "4443:4443"
+   - "8081"
```
Comment (diogomonica) [inline docker-compose.yml]: Is this documented somewhere on the wiki? Which ports are owned by whom?

**Pull request closed without merging**